src/mem/freelist.h

```diff
@@ -188,6 +173,63 @@ namespace snmalloc
       return n;
     }
 
+  public:
+    template<SNMALLOC_CONCEPT(capptr_bounds::c) B>
+    static CapPtr<FreeObject, B> make(CapPtr<void, B> p)
+    {
+      return p.template as_static<FreeObject>();
+    }
+
+    /**
+     * Assign next_object and update its prev_encoded if SNMALLOC_CHECK_CLIENT.
+     * Static so that it can be used on reference to a FreeObject.
+     *
+     * Returns a pointer to the next_object field of the next parameter as an
+     * optimization for repeated snoc operations (in which
+     * next->next_object is nullptr).
+     */
+    template<SNMALLOC_CONCEPT(capptr_bounds::c) B>
+    static CapPtr<FreeObject, B>* store_next(
+      CapPtr<FreeObject, B>* curr,
+      CapPtr<FreeObject, B> next,
+      const FreeListKey& key)
+    {
+      return reinterpret_cast<CapPtr<FreeObject, B>*>(store_next(
```

Comment:
These reinterpret_casts make me nervous, please can you add a comment explaining why they're needed?